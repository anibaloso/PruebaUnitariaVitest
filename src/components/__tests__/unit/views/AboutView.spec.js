import App from '@/App.vue'
import AboutView from '@/views/AboutView.vue'
import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import { createWebHistory, createRouter } from 'vue-router'

describe('AboutView', () => {
    it('Se renderiza AboutView.vue despues de navegar ala ruta "about"', async () => {
        const routerPruebas = createRouter({
            history: createWebHistory(),
            routes: [
                {
                    path: '/about',
                    name: 'about',
                    component: AboutView
                }
            ]
        })

        //navegamos ala ruta /about
        routerPruebas.push({ name: 'about' })
        await routerPruebas.isReady()
        
        //monta el componente app con el router inyectado
        const wrapper = mount(App, {
            global: {
                plugins: [routerPruebas]
            }
        })

        //comprobamos si AboutView esta predente en el renderizado
        expect(wrapper.findComponent(AboutView).exists()).toBeTruthy()
    })

    const wrapper=mount(AboutView)
    it('El HTML se mantiene en las pruebas',()=>{
        expect(wrapper.html()).toMatchSnapshot()
    })
})