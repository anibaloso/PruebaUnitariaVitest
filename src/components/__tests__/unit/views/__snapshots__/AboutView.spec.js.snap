// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutView > El HTML se mantiene en las pruebas 1`] = `
"<div class="about">
  <h1>This is an about page</h1>
</div>"
`;
